client.go

```diff
@@ -240,7 +256,7 @@ func (c *Client) PostJSON(ctx context.Context, endpoint string, data interface{}
 	tracer := c.newTracer()
 	return tracer.Done(c.RoundTrip(func() (*http.Response, error) {
 		data, err := json.Marshal(data)
-		req, err := http.NewRequest(http.MethodPost, endpoint, bytes.NewBuffer(data))
+		req, err := http.NewRequest(method, endpoint, bytes.NewBuffer(data))
```

Comment:
nit: if we used `http.NewRequestContext` we could get rid of the need to use `req.WithContext`